homeassistant/components/mikrotik/device_tracker.py

```diff
@@ -1,251 +1,192 @@
 """Support for Mikrotik routers as device tracker."""
 import logging
+from datetime import timedelta
 
-import ssl
-
-import voluptuous as vol
-
-import homeassistant.helpers.config_validation as cv
-from homeassistant.components.device_tracker import (
+from homeassistant.components.device_tracker import DeviceScanner, SOURCE_TYPE_ROUTER
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.util import slugify
+from homeassistant.const import CONF_HOST, CONF_SCAN_INTERVAL
+from .const import (
     DOMAIN,
-    PLATFORM_SCHEMA,
-    DeviceScanner,
-)
-from homeassistant.const import (
-    CONF_HOST,
-    CONF_PASSWORD,
-    CONF_USERNAME,
-    CONF_PORT,
-    CONF_SSL,
-    CONF_METHOD,
+    CONF_ARP_PING,
+    MIKROTIK_SERVICES,
+    CAPSMAN,
+    WIRELESS,
+    DHCP,
+    ARP,
+    ATTR_DEVICE_TRACKER,
 )
+from . import CONF_METHOD
 
 _LOGGER = logging.getLogger(__name__)
 
-MTK_DEFAULT_API_PORT = "8728"
-MTK_DEFAULT_API_SSL_PORT = "8729"
-
-CONF_LOGIN_METHOD = "login_method"
-MTK_LOGIN_PLAIN = "plain"
-MTK_LOGIN_TOKEN = "token"
-
-CONF_ENCODING = "encoding"
-DEFAULT_ENCODING = "utf-8"
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_METHOD): cv.string,
-        vol.Optional(CONF_LOGIN_METHOD): vol.Any(MTK_LOGIN_PLAIN, MTK_LOGIN_TOKEN),
-        vol.Optional(CONF_PORT): cv.port,
-        vol.Optional(CONF_SSL, default=False): cv.boolean,
-        vol.Optional(CONF_ENCODING, default=DEFAULT_ENCODING): cv.string,
-    }
-)
-
 
-def get_scanner(hass, config):
-    """Validate the configuration and return MTikScanner."""
-    scanner = MikrotikScanner(config[DOMAIN])
-    return scanner if scanner.success_init else None
+def setup_scanner(hass, config, see, discovery_info=None):
```

Comment:
Thank you for that insight. I was hoping to be able to use attributes in the device tracker but I didn't see that in get_scanner. I was using the googlehome bluetooth tracker as an example and saw it was using see and figured it could be used in this one too.

Reply:
We can implement `get_extra_attributes` on the `DeviceScanner` instance.

Reply:
Thanks, I see get_extra_attributes examples. How can I pass config variables to get_scanner? I'm only getting {'device_tracker': {}} from the config variable. Using load_platform 'discovery_info' doesn't seem to pass to get_scanner.

Reply:
No, discovery_info isn't passed to `get_scanner`. You can store the info in hass.data too.

Reply:
I'm confusted. Should there be one device scanner instance running for each Mikrotik host or one total? I was thinking that each host would have it's own device scanner but there is no way to pass the host variable to get_scanner. 

I was creating an api object for each host and storing it in hass.data[MIKROTIK][host] = (api_connection). Does it make sense for one device scanner to connect to each host?

Reply:
It should be possible to set up the platform multiple times, once for each host.